packages/contracts-bedrock/test/L1/SystemConfigInterop.t.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Testing utilities
+import { CommonTest } from "test/setup/CommonTest.sol";
+
+// Libraries
+import { Constants } from "src/libraries/Constants.sol";
+
+// Target contract dependencies
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SystemConfigInterop, NoCode } from "src/L1/SystemConfigInterop.sol";
+import { ConfigType } from "src/L2/L1BlockInterop.sol";
+import { OptimismPortalInterop } from "src/L1/OptimismPortalInterop.sol";
+
+contract SystemConfigWithSetGasPayingToken is SystemConfigInterop {
+    /// @notice External method to set the gas paying token.
+    /// @param _token Address of the token to set as the gas paying token.
+    function setGasPayingToken(address _token) external {
+        _setGasPayingToken(_token);
+    }
+}
+
+contract SystemConfigInterop_Test is CommonTest {
+    /// @notice Marked virtual to be overridden in
+    ///         test/kontrol/deployment/DeploymentSummary.t.sol
+    function setUp() public virtual override {
+        super.setUp();
```

Comment:
I recommend updating the deploy script to follow a pattern similar to [this](https://github.com/ethereum-optimism/optimism/blob/14151dbcc8f5ee1fea68052b9d2d503a061fc2b6/packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol#L42) or [this](https://github.com/ethereum-optimism/optimism/blob/14151dbcc8f5ee1fea68052b9d2d503a061fc2b6/packages/contracts-bedrock/test/L1/DataAvailabilityChallenge.t.sol#L16)

Reply:
You could use https://github.com/ethereum-optimism/protocol-quest/issues/247

Reply:
created the following PRs implementing this logic:
- [SystemConfigInterop](https://github.com/ethereum-optimism/optimism/pull/10581)
- [L1BlockInterop](https://github.com/ethereum-optimism/optimism/pull/10582)
- [OptimismPortalInterop](https://github.com/ethereum-optimism/optimism/pull/10583)